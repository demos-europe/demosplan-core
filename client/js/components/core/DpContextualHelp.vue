<license>
  (c) 2010-present DEMOS E-Partizipation GmbH.

  This file is part of the package demosplan,
  for more information see the license file.

  All rights reserved
</license>

<documentation>
  <!--

  The DpContextualHelp component renders an icon indicating contextual help,
  which shows a tooltip able to handle text or html content.

  -->
  <usage>
    <dp-contextual-help :text="Translator.trans('statements.import.hint')" />
  </usage>
</documentation>

<script>
export default {
  name: 'DpContextualHelp',

  functional: true,

  props: {
    /**
     * A translation key representing the actual tooltip content.
     * May actually be any string including html.
     */
    text: {
      type: String,
      required: true
    }
  },

  render: function (h, { props, data }) {
    return h('i', {
      attrs: {
        class: `fa fa-question-circle ${typeof data.staticClass === 'undefined' ? '' : data.staticClass}`,
        'aria-label': Translator.trans('contextual.help')
      },
      directives: [
        {
          name: 'tooltip',
          value: props.text
        }
      ]
    })
  }
}
</script>
