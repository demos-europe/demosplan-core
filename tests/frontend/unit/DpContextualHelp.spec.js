/**
 * (c) 2010-present DEMOS E-Partizipation GmbH.
 *
 * This file is part of the package demosplan,
 * for more information see the license file.
 *
 * All rights reserved
 */

import { createLocalVue } from '@vue/test-utils'
import DpContextualHelp from '@DemosPlanCoreBundle/components/DpContextualHelp'
import shallowMountWithGlobalMocks from '@DemosPlanCoreBundle/VueConfigLocal'

describe('DpContextualHelp', () => {
  it('should be an object', () => {
    expect(typeof DpContextualHelp).toBe('object')
  })

  it('should be named DpContextualHelp', () => {
    expect(DpContextualHelp.name).toBe('DpContextualHelp')
  })

  it('should render the correct html', async () => {
    const localVue = createLocalVue()

    const instance = shallowMountWithGlobalMocks(DpContextualHelp, {
      propsData: {
        text: 'This is the tooltip content.'
      },
      localVue
    })

    expect(instance.html()).toMatchSnapshot()
  })
})
