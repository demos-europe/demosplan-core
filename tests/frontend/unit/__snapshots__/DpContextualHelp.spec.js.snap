// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DpContextualHelp should render the correct html 1`] = `"<i class=\\"fa fa-question-circle  has-tooltip\\" aria-label=\\"contextual.help\\" data-original-title=\\"null\\"></i>"`;
